models/aeif_psc_delta.cpp

```diff
@@ -0,0 +1,536 @@
+    names::w, &aeif_psc_delta::get_y_elem_< aeif_psc_delta::State_::W > );
+}
+}
+
+
+extern "C" int
+nest::aeif_psc_delta_dynamics( double,
+  const double y[],
+  double f[],
+  void* pnode )
+{
+  // a shorthand
+  typedef nest::aeif_psc_delta::State_ S;
+
+  // get access to node so we can almost work as in a member function
+  assert( pnode );
+  const nest::aeif_psc_delta& node =
+    *( reinterpret_cast< nest::aeif_psc_delta* >( pnode ) );
+
+  // y[] here is---and must be---the state vector supplied by the integrator,
+  // not the state vector in the node, node.S_.y[].
+
+  // The following code is verbose for the sake of clarity. We assume that a
+  // good compiler will optimize the verbosity away ...
+
+  // shorthand for state variables
+  const bool is_refractory = node.S_.r_ > 0;
+
+  const double V = is_refractory
+    ? node.P_.V_reset_
+    : std::min( y[ S::V_M ], node.P_.V_peak_ ); // bind V to the
+  // USER DEFINED V_peak_ value in Parameters.
+  const double& w = y[ S::W ];
+
+  const double I_spike = node.P_.Delta_T == 0. ? 0. : node.P_.g_L
+      * node.P_.Delta_T * std::exp( ( V - node.P_.V_th ) / node.P_.Delta_T );
+
+  // dv/dt
+  f[ S::V_M ] =
+    is_refractory ? 0.0 : ( -node.P_.g_L * ( V - node.P_.E_L ) + I_spike - w
+                            + node.P_.I_e + node.B_.I_stim_ ) / node.P_.C_m;
+
+  // Adaptation current w.
+  f[ S::W ] = ( node.P_.a * ( V - node.P_.E_L ) - w ) / node.P_.tau_w;
+
+  return GSL_SUCCESS;
+}
+
+/* ----------------------------------------------------------------
+ * Default constructors defining default parameters and state
+ * ---------------------------------------------------------------- */
+
+nest::aeif_psc_delta::Parameters_::Parameters_()
+  : V_peak_( 0.0 )    // mV
+  , V_reset_( -60.0 ) // mV
+  , t_ref_( 0.0 )     // ms
+  , g_L( 30.0 )       // nS
+  , C_m( 281.0 )      // pF
+  , E_L( -70.6 )      // mV
+  , Delta_T( 2.0 )    // mV
+  , tau_w( 144.0 )    // ms
+  , a( 4.0 )          // nS
+  , b( 80.5 )         // pA
+  , V_th( -50.4 )     // mV
+  , I_e( 0.0 )        // pA
+  , gsl_error_tol( 1e-6 )
+{
+}
+
+nest::aeif_psc_delta::State_::State_( const Parameters_& p )
+  : r_( 0 )
+{
+  y_[ 0 ] = p.E_L;
+  for ( size_t i = 1; i < STATE_VEC_SIZE; ++i )
+    y_[ i ] = 0;
+}
+
+nest::aeif_psc_delta::State_::State_( const State_& s )
+  : r_( s.r_ )
+{
+  for ( size_t i = 0; i < STATE_VEC_SIZE; ++i )
+    y_[ i ] = s.y_[ i ];
+}
+
+nest::aeif_psc_delta::State_& nest::aeif_psc_delta::State_::operator=(
+  const State_& s )
+{
+  assert( this != &s ); // would be bad logical error in program
+
+  for ( size_t i = 0; i < STATE_VEC_SIZE; ++i )
+    y_[ i ] = s.y_[ i ];
+  r_ = s.r_;
+  return *this;
+}
+
+/* ----------------------------------------------------------------
+ * Paramater and state extractions and manipulation functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::aeif_psc_delta::Parameters_::get( DictionaryDatum& d ) const
```

Comment:
`refractory_input` is already in `names`, it just needs to be used here, and also in `iaf_psc_delta`.